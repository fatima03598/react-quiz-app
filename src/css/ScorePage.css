.scorepage-container {
    margin: 3em;
    display: flex;
    text-align: center;
    flex-direction: column;
    height: 100%;
}

.scores-container {
    margin: 1em;
    display: flex;
    flex-direction: column;
    text-align: center;
    justify-content: space-evenly;
}