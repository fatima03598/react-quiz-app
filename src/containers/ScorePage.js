import React from 'react';
import '../css/ScorePage.css';

function ScorePage(props) {
    return(
        <div>
            <h1>Scores Page</h1>
            <div className="scores-container">
                    <h2>Scores:</h2>
                    {(Object.values(props.location.state.stateFinal.userNames)).map(user => {
                        return (<h3>{user}:{props.location.state.userScores[user]}</h3>)
                    })}
                </div>
        </div>
    );
}

export default ScorePage;