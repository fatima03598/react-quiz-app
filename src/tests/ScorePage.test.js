import React from 'react';
import { BrowserRouter,  MemoryRouter, Link } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import ScorePage from '../containers/ScorePage';
import renderer from 'react-test-renderer';


describe('ScorePage', () => {

    it('matches the snapshot', () => {
      const tree = renderer.create(<ScorePage location={{userScores: {1: 0}, state:{userScores: {1: 0}, stateFinal: {userNames: {0: "1"}}}}}/>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render scores-container div', () => {
        const wrapper = mount( <ScorePage location={{userScores: {1: 0}, state:{userScores: {1: 0}, stateFinal: {userNames: {0: "1"}}}}}/> )
        expect(wrapper.exists('.scores-container')).toEqual(true);
        });

    it('should render a h2', () => {
      const wrapper = mount( <ScorePage location={{userScores: {1: 0, 2: 0}, state:{userScores: {1: 0, 2: 0}, stateFinal: {userNames: {0: "1", 1: "2"}}}}}/> )
      expect(wrapper.exists('h2')).toEqual(true);
      });
    
    it('should render a h2', () => {
      const wrapper = mount( <ScorePage location={{userScores: {1: 1, 2: 0}, state:{userScores: {1: 1, 2: 0}, stateFinal: {userNames: {0: "1", 1: "2"}}}}}/> )
      expect(wrapper.exists('h2')).toEqual(true);
      });
});
