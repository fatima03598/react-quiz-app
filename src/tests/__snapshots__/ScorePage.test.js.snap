// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScorePage matches the snapshot 1`] = `
<div
  className="scorepage-container"
>
  <h1>
    Final Scores
  </h1>
  <div
    className="scores-container"
  >
    <h2>
       Good Job!
    </h2>
    <h3>
      1
      :   
      0
    </h3>
  </div>
</div>
`;
